airflow/contrib/operators/gcs_to_bq.py

```diff
@@ -248,7 +252,7 @@ def execute(self, context):
                 time_partitioning=self.time_partitioning,
                 cluster_fields=self.cluster_fields)
 
-        if self.max_id_key:
+        if self.do_xcom_push and self.max_id_key:
```

Comment:
I'm not really sure if the `do_xcom_push` flag adds a lot of value here.

Reply:
I wondered too but I decided to add `xcom_push` flag so that the `max_id_key` is not an exceptional flag to enable xcom feature. The best might be renaming `max_id_key` to `xcom_push` flag & add documenation to explain what is pushed to xcom when the feature is enabled.